src/Components/ArtworkAlert/Hooks/useArtworkAlertContext.tsx

```diff
@@ -0,0 +1,208 @@
+import {
+  FC,
+  createContext,
+  useRef,
+  useContext,
+  useReducer,
+  useEffect,
+  useState,
+} from "react"
+
+import {
+  SearchCriteriaAttributeKeys,
+  SearchCriteriaAttributes,
+} from "Components/SavedSearchAlert/types"
+import { useCreateSavedSearch } from "Utils/Hooks/Mutations/useCreateSavedSearchMutation"
+import { getAllowedSearchCriteria } from "Components/SavedSearchAlert/Utils/savedSearchCriteria"
+
+type Settings = {
+  details: string
+  email: boolean
+  name: string
+  push: boolean
+}
+
+export type State = {
+  settings: Settings
+  criteria: SearchCriteriaAttributes
+  searchCriteriaID?: string
+  currentArtworkID?: string
+}
+
+export const DEFAULT_STATE: State = {
+  settings: {
+    name: "",
+    details: "",
+    email: true,
+    push: false,
+  },
+  criteria: {},
+}
+
+type Action =
+  | { type: "RESET" }
+  | { type: "SET_SETTINGS"; payload: Settings }
+  | { type: "SET_CRITERIA"; payload: SearchCriteriaAttributes }
+  | {
+      type: "SET_CRITERIA_ATTRIBUTE"
+      payload: {
+        key: SearchCriteriaAttributeKeys
+        value: string | string[] | number | boolean | null
+      }
+    }
+  | {
+      type: "REMOVE_CRITERIA_ATTRIBUTE_VALUE"
+      payload: {
+        key: SearchCriteriaAttributeKeys
+        value: string | string[] | number | boolean | null
+      }
+    }
+  | { type: "SET_SEARCH_CRITERIA_ID"; payload: string }
```

Comment:
**walkthrough**: Also inspired by the onboarding setup, this represents the actions available via dispatch from context consumers. Useful for responding to UX interactions in forms / filters.